### DISCUSSION TITLE
测试数据存储：单文件/压缩包

### DISCUSSION THREAD

**Discussion by undefined-moe** [hydro-dev/Hydro#36] [General]
使用压缩包可减少存储消耗，同时便于传输
但按测试点下载数据需要存储为单文件/同时Hack此类动态patch测试包的功能也倾向于使用单文件存储。

**Comment by junukwon7** [comment]
> The use of compressed packages can reduce storage consumption and facilitate transmission.
> However, downloading data by test points needs to be stored as a single file/At the same time, the function of Hack such dynamic patch test package also tends to use single file storage.

Are testcases not unzipped after uploading it? i.e, cms-dev supports both single upload and compressed packages because it makes easier to upload. All testcases are stored and managed separately.
(did I get it wrong?)

**Comment by undefined-moe** [comment]
yes, now the data stores as a single zipfile in the database.(as web server doesnt need to handle these file before), and unzipped at judge-client side.  
i'm also thinking about changing the storage engine to s3 compatible(like min.io)

**Comment by junukwon7** [comment]
> yes, now the data stores as a single zipfile in the database.(as web server doesnt need to handle these file before), and unzipped at judge-client side.
> i'm also thinking about changing the storage engine to s3 compatible(like min.io)

Does the judge client unzip the file for each submission?
p.s. did you check out the new discussion page for GitHub?

**Comment by undefined-moe** [comment]
No, it's unzipped and cached into ~/.cache/hydro at first submission (or first submission after the testdata updated).

**Comment by undefined-moe** [comment]
Now its unzipped and stored. (branch master, not released yet)

**Comment by undefined-moe** [comment]
New version released. To perform this update:

- Download and install [MinIO](https://min.io)
- Set environment variable `MINIO_ACCESS_KEY` and `MINIO_SECRET_KEY`
- upgrade and restart hydrooj

The upgrade process should be starting automaticly (better to start from a terminal instead of using process manager when upgrading). If not, check configuration in control panel, update it and restart again.
